CMakeLists.txt

```diff
@@ -27,6 +27,78 @@ ELSE()
     SET(VER_BUILD 0)
 ENDIF()
 
+# Configuration of compilers
+## Compiler flags to add
+### Flags for GCC and Clang
+set(GNU_FLAGS_WARNING_YES "-Wall -Wextra -Wformat=2 -Wpedantic -Wshadow -Wmissing-declarations -Wfloat-equal")
```

Comment:
Second thing or first thing, the gnu_flags_warnings yes no separation looks silly. Please use only one variable, otherwise it's even less readable for no added value.